src/Tgstation.Server.Host/Program.cs

```diff
@@ -67,6 +70,34 @@ public static async Task<int> Main(string[] args)
 				args = listArgs.ToArray();
 			}
 
+			if (InteropServices.RuntimeInformation.IsOSPlatform(InteropServices.OSPlatform.Linux))
+			{
+				using var proc = new Process
+				{
+					StartInfo = new ProcessStartInfo
+					{
+						FileName = "id",
+						Arguments = "-u",
+						UseShellExecute = false,
+						RedirectStandardOutput = true,
+						CreateNoWindow = true,
+					},
+				};
+
+				proc.Start();
+				await proc.WaitForExitAsync();
+				if (proc.ExitCode is not 0 || !int.TryParse(await proc.StandardOutput.ReadToEndAsync(), out var uid))
+				{
+					Console.Error.WriteLine("Failed to obtain user id.");
+					return 1;
+				}
+
+				if (uid is 0)
+				{
+					Console.Error.WriteLine("TGS is being run as root. This is not recommended and may prevent launch in a future version.");
```

Comment:
This is not the place to put arbitrary startup checks. There's a function called `CheckSystemCompatibility` somewhere. Use that instead.

Reply:
Should also be "as the root account".

Reply:
These need to use `ILogger.LogWarning` instead of direct console output.